src/dev/build/tasks/os_packages/docker_generator/templates/Dockerfile

```diff
@@ -0,0 +1,122 @@
+################################################################################
+# This Dockerfile was generated from the template at:
+#   src/dev/build/tasks/os_packages/docker_generator/templates/Dockerfile
+#
+# Beginning of multi stage Dockerfile
+################################################################################
+
+################################################################################
+# Build stage 0 `builder`:
+# Extract Kibana artifact
+################################################################################
+FROM {{{baseOSImage}}} AS builder
+
+{{#ubi}}
+RUN {{packageManager}} install -y findutils tar gzip
+{{/ubi}}
+
+{{#usePublicArtifact}}
+RUN cd /opt && \
+  curl --retry 8 -s -L -O https://artifacts.elastic.co/downloads/kibana/{{artifactTarball}} && \
+  cd -
+{{/usePublicArtifact}}
+
+{{^usePublicArtifact}}
+COPY {{artifactTarball}} /opt
+{{/usePublicArtifact}}
+
+RUN mkdir /usr/share/kibana
+WORKDIR /usr/share/kibana
+RUN tar --strip-components=1 -zxf /opt/{{artifactTarball}}
+# Ensure that group permissions are the same as user permissions.
+# This will help when relying on GID-0 to run Kibana, rather than UID-1000.
+# OpenShift does this, for example.
+# REF: https://docs.openshift.org/latest/creating_images/guidelines.html
+RUN chmod -R g=u /usr/share/kibana
+RUN find /usr/share/kibana -type d -exec chmod g+s {} \;
+
+################################################################################
+# Build stage 1 (the actual Kibana image):
+#
+# Copy kibana from stage 0
+# Add entrypoint
+################################################################################
+FROM {{{baseOSImage}}}
+EXPOSE 5601
+
+RUN for iter in {1..10}; do \
+      {{packageManager}} update --setopt=tsflags=nodocs -y && \
+      {{packageManager}} install --setopt=tsflags=nodocs -y \
```

Comment:
does this need to support apt or just dnf/yum?  asking by the setopt flag

Reply:
Yeah, just `dnf` and `yum`.